sdks/typescript/__tests__/integration.test.ts

```
@@ -0,0 +1,418 @@
+      expect(firstPage.total).toBe(4);
+      expect(firstPage.size).toBe(2);
+      expect(firstPage.hasNextPage).toBe(true);
+
+      // Step 2: Get data from first page
+      const firstPageData_ = await firstPage.data();
+      expect(firstPageData_).toHaveLength(2);
+
+      // Step 3: Get next page
+      const secondPage = await firstPage.nextPage();
+      expect(secondPage).not.toBeNull();
+      expect(secondPage!.hasNextPage).toBe(false);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)